Source/Scene/PostProcessBloomStage.js

```diff
@@ -0,0 +1,144 @@
+define([
+        '../Core/Check',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Shaders/PostProcessFilters/BloomComposite',
+        '../Shaders/PostProcessFilters/ContrastBias',
+        './PostProcess',
+        './PostProcessBlurStage',
+        './PostProcessComposite'
+    ], function(
+        Check,
+        defineProperties,
+        destroyObject,
+        BloomComposite,
+        ContrastBias,
+        PostProcess,
+        PostProcessBlurStage,
+        PostProcessComposite) {
+    'use strict';
+
+    /**
+     * Post process stage for bloom. Implements {@link PostProcess}.
+     *
+     * @alias PostProcessBloomStage
+     * @constructor
+     *
+     * @private
+     */
+    function PostProcessBloomStage() {
+        this._name = 'czm_bloom';
+
+        this._contrastBias = new PostProcess({
+            name : 'czm_bloom_contrast_bias',
+            fragmentShader : ContrastBias,
+            uniformValues : {
+                contrast : 0.0,
+                brightness : 0.0
+            }
+        });
+        this._blur = new PostProcessBlurStage({
+            name : 'czm_bloom_blur'
+        });
+        var generateComposite = new PostProcessComposite({
+            name : 'czm_bloom_contrast_bias_bloom',
+            processes : [this._contrastBias, this._blur]
+        });
+
+        var bloomComposite = new PostProcess({
+            name : 'czm_bloom_generate_composite',
+            fragmentShader : BloomComposite,
```

Comment:
The naming is a bit confusing - below the composite post process is created, but here this is also a composite.  I think this is compositing the previous stages, and below is the overall composite to bring together the stages, but I think this the names could be improved.

Perhaps if we go with the factory approach, this will not matter.  Probably worth checking the architecture for Babylon and three if you haven't already.